### PR TITLE
feat: add support for 'finalized' revision in accounts

### PR DESCRIPTION
**Pull request by darrenvechain**
# Description

This PR allows clients to specify `finalized` as a revision on the accounts endpoints.

The solo instance is also modified to return the best block as the finalized block

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [x] `accounts_test.go`
- [x]  Manually testing a thor solo instance
- [x]  Manually @  https://mainnet.blue.dev.node.vechain.org/doc/stoplight-ui/#/paths/accounts-address/get

```bash
curl --request GET \
  --url 'https://mainnet.blue.dev.node.vechain.org/accounts/0x3504c0089Ce44d5709BB97a957EC40C76F18dCdA?revision=best' \
  --header 'Accept: application/json, text/plain'
  
  
curl --request GET \
  --url 'https://mainnet.blue.dev.node.vechain.org/accounts/0x3504c0089Ce44d5709BB97a957EC40C76F18dCdA?revision=finalized' \
  --header 'Accept: application/json, text/plain'
```

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] New and existing E2E tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have not added any vulnerable dependencies to my code

### CODE REVIEW

**Comment by nwbrettski** [comment]
Add finalized to transactions endpoint.

Add documentation clarifying what finalized means:
- 360 blocks
- 2 rounds of BFT

File: cmd/thor/solo/types.go [hunk 1]
```diff
@@ -22,15 +22,15 @@ func (comm *Communicator) PeersStats() []*comm.PeerStats {
 
 // BFTEngine is a fake bft engine for solo.
 type BFTEngine struct {
-	finalized thor.Bytes32
+	repo *chain.Repository
 }
 
 func (engine *BFTEngine) Finalized() thor.Bytes32 {
-	return engine.finalized
+	return engine.repo.BestBlockSummary().Header.ID()
```
[the remark this answers is not in the document]
Comment (libotony) [reply cmd/thor/solo/types.go:29]: We can add a ticket to add a simulated BFT to solo.

[the remark this answers is not in the document]
Comment (darrenvechain) [reply cmd/thor/solo/types.go:29]: I think it would be good to simulate it. Could we return best block -1 or something?

**Comment by libotony** [comment]
> Add finalized to:
> 
> * transactions endpoint
> * debug trace call endpoint
> 
> Add documentation clarifying what finalized means:
> 
> * 360 blocks
> * 2 rounds of BFT

Had a discussion with @darrenvechain, transaction's endpoint does not need `finalized`, it used to only accept `blockID or null`

**Comment by libotony** [comment]
Do not forget to sync the version `thor.yaml` with `cmd/thor/VERSION` @darrenvechain

File: api/debug/debug.go [hunk 2]
```diff
@@ -174,10 +177,13 @@ func (d *Debug) handleTraceCall(w http.ResponseWriter, req *http.Request) error
...
-
-	summary, err := d.handleRevision(req.URL.Query().Get("revision"))
+	revision, err := utils.ParseRevision(req.URL.Query().Get("revision"))
 	if err != nil {
-		return err
+		return utils.BadRequest(errors.WithMessage(err, "revision"))
+	}
+	summary, err := utils.GetSummary(revision, d.repo, d.bft)
+	if err != nil {
+		return utils.BadRequest(errors.WithMessage(err, "revision"))
```
Comment (libotony) [inline api/debug/debug.go:186]:
```suggestion
		if a.repo.IsNotFound(err) {
			return utils.BadRequest(errors.WithMessage(err, "revision"))
		}
		return err
```

File: api/debug/debug.go [hunk 3]
```diff
@@ -174,9 +177,15 @@ func (d *Debug) handleTraceCall(w http.ResponseWriter, req *http.Request) error
...
 	}
-
-	summary, err := d.handleRevision(req.URL.Query().Get("revision"))
+	revision, err := utils.ParseRevision(req.URL.Query().Get("revision"))
+	if err != nil {
+		return utils.BadRequest(errors.WithMessage(err, "revision"))
+	}
+	summary, err := utils.GetSummary(revision, d.repo, d.bft)
 	if err != nil {
+		if a.repo.IsNotFound(err) {
```
Comment (libotony) [inline api/debug/debug.go:186]:
```suggestion
		if d.repo.IsNotFound(err) {
```
😢

**Review by libotony — APPROVED** [review]

**Pull request merged**